README.md

```diff
@@ -0,0 +1,141 @@
+<div style="text-align:center">
+    <h1> Cloudlet</h1>
+    <p>The almost fast FaaS</p>
+    <img src="./assets/demo.gif" alt="Demo" />
+</div>
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Prerequisites](#prerequisites)
+- [Run Locally](#run-locally)
+  - [Clone the project](#clone-the-project)
+  - [Setup](#setup)
+  - [Start the VMM](#start-the-vmm)
+  - [Run the API](#run-the-api)
+  - [Send the request using the cli](#send-the-request-using-the-cli)
+- [Architecture](#architecture)
+- [Config file](#config-file)
+
+## Prerequisites
+
+Basic dependencies (with apt):
+
+```bash
+apt install build-essential cmake pkg-config libssl-dev flex bison libelf-dev
+```
+
+Others:
+
+- add rust musl target: `rustup target add x86_64-unknown-linux-musl`
+- add protoc binary: see [`github.com/protocolbuffers/protobuf`](https://github.com/protocolbuffers/protobuf)
+- [just](https://github.com/casey/just): `cargo install just`
+
+## Run Locally
+
+### Clone the project
+
+```bash
+git clone https://github.com/virt-do/cloudlet
+```
+
+### Setup
+
+Go to the project directory:
+
+```bash
+cd cloudlet
+```
+
+Create a TOML config file or update the [existing one](./src/agent/examples/config.toml):
+
+```bash
+cat << EOF > src/agent/examples/config.toml
+workload-name = "fibonacci"
+language = "rust"
+action = "prepare-and-run"
+
+[server]
+address = "localhost"
+port = 50051
+
+[build]
+source-code-path = "$(readlink -f ./src/agent/examples/main.rs)"
+release = true
+EOF
+```
+
+Make sure to update the source-code-path to the path of the source code you want to run.
+Use an absolute path.
+
+[Here](#config-file) are more informations about each field
+
+### Start the VMM
+
+> [!WARNING]
+> Make sure to replace $CARGO_PATH with the path to your cargo binary
+> 
+> ```bash
+> export CARGO_PATH=$(which cargo)
+> ```
+
+```bash
+sudo -E capsh --keep=1 --user=$USER --inh=cap_net_admin --addamb=cap_net_admin -- -c  'RUST_BACKTRACE=1 '$CARGO_PATH' run --bin vmm -- grpc'
+```
+
+### Run the API
+
+```bash
+cargo run --bin api
+```
+
+### Send the request using the cli 
+
+```bash
+cargo run --bin cli run --config-path src/agent/examples/config.toml
```

Comment:
No the CLI got a command named run but this doesn't make much sense since it's the only command available. The stop one is not needed as a command since it will be send as a signal ctrl + c or ctrl + d

Reply:
Okay understood. Don't you need the `--` in any case tho'?

Reply:
I've tried and apparently you don't need it

But maybe it's better to include it, to show the separation

Reply:
adding it just in case